twingate/internal/client/service-account.go

```diff
@@ -72,6 +72,19 @@ func (client *Client) UpdateServiceAccount(ctx context.Context, serviceAccount *
 }
 
 func (client *Client) DeleteServiceAccount(ctx context.Context, serviceAccountID string) error {
+	serviceAccount, err := client.ReadServiceAccount(ctx, serviceAccountID)
+	if err != nil && !errors.Is(err, ErrGraphqlResultIsEmpty) {
+		return err
+	}
+
+	if serviceAccount != nil {
+		for _, key := range serviceAccount.Keys {
+			if err := client.RevokeServiceKey(ctx, key); err != nil {
```

Comment:
Not sure I understand the process here.
The Service Account __owns__ its keys so if someone deleted it - the keys are gone too. 
There's no need to make API calls to revoke them...

Reply:
Also mutation would error in this case...